docs/contributor/arch/013-resource-attributes-enrichment-for-non-workload-metrics.md

```diff
@@ -0,0 +1,60 @@
+# 2. Resource Attributes enrichment for non-workload metrics
+
+Date: 2024-08-30
+
+## Status
+
+Accepted
+
+## Context
+
+Currently, the [k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/README.md) is used in the Metric Gateway to enrich metrics resource attributes with k8s metadata.
+This is very useful for workload metrics (e.g. Pod and container metrics). However, for non-workload metrics (e.g. node metrics), the metrics will be incorrectly associated with Metric Agent Pod, since the Metric Agent Pod is the one that emits these metrics.
+
+For example, in the diagram shown below, the `k8s.node.cpu.usage`metric is incorrectly associated with the `telemetry-metric-agent-4ncx9` pod. This creates confusion, as the user might incorrectly think that this metric provides the node CPU consumed by the `telemetry-metric-agent-4ncx9` pod. While in reality, it provides the entire node CPU usage.
+In addition, the metrics from the system namespaces (including `kyma-system` namespace) are excluded by default. Thus, this metric will be dropped by default even if the user enables the collection of node metrics.
+
+![Node Metric With k8sattributes Processor](../assets/node-metric-with-k8sattributes-processor.png)
+
+
+## Decision
+
+There are different possible solutions to solve this problem:
+
+**Option 1: Using connectors**
+
+![Connectors](../assets/connectors.drawio.svg)
+
+We can split our MetricPipeline into 3 sub-pipelines which are connected together using [Connectors](https://opentelemetry.io/docs/collector/building/connector/) as shown in the diagram above.
+A `Receiver Pipeline` which contains the receivers components.
+A `k8sattribute Processor` which contains only the k8sattributes processor.
+A `Processor + Exporter Pipeline` which contains the rest of the processors and the exporters. The metrics will be routed to the `k8sattribute Processor` only if they are workload metrics.
+Otherwise, the metrics will bypass the `k8sattribute Processor` and will be routed directly to the `Processor + Exporter Pipeline`.
```

Comment:
```suggestion
Otherwise, the metrics will bypass the `k8sattribute processor` and will be routed directly to the `Processor + Exporter Pipeline`.
```

Reply:
I would prefer to keep the `k8sattribute Processor Pipeline` capitalized, because it represents the entity in the diagram